### ISSUE TITLE
Intl - polyfills  contributions

### ISSUE THREAD

**Issue by romulocintra** [tc39/ecma402#700]
Placeholder issue to organize Polyfills creation/contributions.

#### Get Started
- Consult existing Polyfills in the [project wiki](https://github.com/tc39/ecma402/wiki/Proposal-and-PR-Progress-Tracking#stage-3), those with :x:  means that are missing or not started 

![image](https://user-images.githubusercontent.com/1572026/176629463-34c2b56d-8459-4d33-9fc5-0ff41df105f9.png)

- Choose the one you want to contribute/work on then create an issue in the proposal repo then you can update the [project wiki page](https://github.com/tc39/ecma402/wiki/Proposal-and-PR-Progress-Tracking#stage-3) to reflect the status

![image](https://user-images.githubusercontent.com/1572026/176630507-cf6fb536-de0f-41e5-8da1-59bccc030f43.png)

#### Recommended projects to add/contribute with Polyfills

- https://github.com/formatjs/formatjs 
- Add yours in the comments...

**Comment by romulocintra** [comment]
cc @micheleriva @UlisesGascon

**Comment by micheleriva** [comment]
@romulocintra by looking at the project wiki ([https://github.com/tc39/ecma402/wiki/Proposal-and-PR-Progress-Tracking#stage-4-for-2022-edition](https://github.com/tc39/ecma402/wiki/Proposal-and-PR-Progress-Tracking#stage-4-for-2022-edition)), I see we're missing the following polyfill too:

## Stage 4
- [Intl.Segmenter](https://github.com/tc39/proposal-intl-segmenter)
- [Extend TimeZoneName Option](https://github.com/tc39/proposal-intl-extend-timezonename)
- [Intl.DisplayNames v2](https://github.com/tc39/intl-displaynames-v2)

## Stage 3
- [Intl.NumberFormat V3](https://github.com/tc39/proposal-intl-numberformat-v3)
- [Intl.DurationFormat](https://github.com/tc39/proposal-intl-duration-format)

Are we interested in proceeding with Stage 4 polyfill too?

**Comment by longlho** [comment]
apologies for not keeping up w/ the polyfills. `formatjs` is def open to polyfill contributions! The stage 3/4 polyfills are certainly do-able, except `Intl.Segmenter` which imo is kinda impossible to polyfill and dataset might be massive (much more than `Intl.DateTimeFormat` which contains both CLDR & IANA dataset).

**Comment by micheleriva** [comment]
@longlho as you might have noticed in the `formatjs` repo, I'm working to provide a polyfill for all the open proposals... we'll see what can we do with  `Intl.Segmenter`, I agree that it might be a bit more complicated to optimize

**Comment by longlho** [comment]
no, the hardest part of writing Intl polyfills is synthesizing data sources from CLDR

**Comment by romulocintra** [comment]
> @romulocintra by looking at the project wiki (https://github.com/tc39/ecma402/wiki/Proposal-and-PR-Progress-Tracking#stage-4-for-2022-edition), I see we're missing the following polyfill too:
> 
> ## Stage 4
> * [Intl.Segmenter](https://github.com/tc39/proposal-intl-segmenter)
> * [Extend TimeZoneName Option](https://github.com/tc39/proposal-intl-extend-timezonename)
> * [Intl.DisplayNames v2](https://github.com/tc39/intl-displaynames-v2)
> 
> ## Stage 3
> * [Intl.NumberFormat V3](https://github.com/tc39/proposal-intl-numberformat-v3)
> * [Intl.DurationFormat](https://github.com/tc39/proposal-intl-duration-format)
> 
> Are we interested in proceeding with Stage 4 polyfill too?

Hi, @micheleriva do you have any plans for DurationFormat Polyfill?

**Comment by micheleriva** [comment]
Hi @romulocintra, we might have found a way to polyfill [Intl.Segmenter](https://github.com/tc39/proposal-intl-segmenter) efficiently. ETA is one week of work.

If you find `DurationFormat` to be a priority, let me know so we can focus here!

**Comment by micheleriva** [comment]
@romulocintra as per the private message, we can take charge of `DurationFormat` polyfill. I'll update y'all ASAP

**Comment by romulocintra** [comment]
> @romulocintra as per the private message, we can take charge of `DurationFormat` polyfill. I'll update y'all ASAP

This is Fantastic thank you @micheleriva  and team :)  🙏  🙏  🙏  🙏  🙏  🙏


cc @littledan @sffc

**Comment by eemeli** [comment]
For Intl.NumberFormat v3, [`intl-pluralrules@next`](https://www.npmjs.com/package/intl-pluralrules/v/next) should be a spec-compliant polyfill for the Intl.PluralRules object, in particular including the added `selectRange()` method.

**Comment by sffc** [comment]
There's some progress on the Intl.NumberFormat v3 polyfill in https://github.com/formatjs/formatjs/pull/3860. CC @longlho

**Comment by longlho** [comment]
Yeah that got merged

**Comment by HadrienGardeur** [comment]
> […] we might have found a way to polyfill [Intl.Segmenter](https://github.com/tc39/proposal-intl-segmenter) efficiently. ETA is one week of work.

@micheleriva was this work on an `Intl.Segmenter` polyfill ever completed?
